org.opent2t.sample.thermostat.superpopular/com.insteon.thermostat/js/thingTranslator.js

```diff
@@ -263,6 +265,178 @@ function validateResourceGet(resourceId) {
+        targetLow = providerSchema.cool_point;
+    } else if (providerSchema.hasOwnProperty('heat_point')) {
+        targetHigh = providerSchema.heat_point;
+        targetLow = providerSchema.heat_point;
+    } else {
+        // Thermostat is not set to anything
+    }
+
+    var range = targetHigh - targetLow;
+    var halfRange = Math.round(range / 2);
+    var newTargetLow = temperature - halfRange;
+    var newTargetHigh = temperature + halfRange;
+
+    // If either value is outside min/max range, adjust accordingly
+    if (newTargetLow < min) {
+        newTargetLow = min;
+        newTargetHigh = min + range;
+    }
+
+    if (newTargetHigh > max) {
+        newTargetHigh = max;
+        newTargetLow = Math.max(min, max - range);
+    }
+
+    var adjustedRange = newTargetLow - targetLow;
+
+    // Temp, just for the response - convert back to users units
+    response.cool_point = convertTemperature(newTargetHigh, providerUnit, unit);
+    response.heat_point = convertTemperature(newTargetLow, providerUnit, unit);
+    response.mode = 'auto';
+    response.unit = unit;
+
+    // Command for insteon
+    command.command = adjustedRange < 0 ? 'temp_down' : 'temp_up';
+    command.temp = Math.abs(adjustedRange);
+
+    return { response, command };
+}
+
+function getTargetTemperatureHigh(resourceSchema, providerSchema) {
+    var response = {};
+    var command = {}
+
+    var unit = getValidatedUnit(resourceSchema);
+    var providerUnit = getUnitSafe(providerSchema, unit);
+    
+    // Temp just for the resource response
+    response.cool_point = resourceSchema.temperature;
+    response.mode = 'cool';
+    response.unit = unit;
+
+    // Command for insteon
+    command.command = 'set_cool_to';
+    command.temp = convertTemperature(resourceSchema.temperature, unit, providerUnit);
+
+    return { response, command };
+}
+
+function getTargetTemperatureLow(resourceSchema, providerSchema) {
+    var response = {};
+    var command = {}
+
+    var unit = getValidatedUnit(resourceSchema);
+    var providerUnit = getUnitSafe(providerSchema, unit);
+
+    // Temp just for the resource response
+    response.heat_point = resourceSchema.temperature;
+    response.mode = 'heat';
+    response.unit = unit;
+
+    // Command for insteon
+    command.command = 'set_heat_to';
+    command.temp = convertTemperature(resourceSchema.temperature, unit, providerUnit);
+
+    return { response, command };
+}
+
+function getUnitSafe(providerSchema, defaultUnit) {
+    return providerSchema.hasOwnProperty('unit') && providerSchema.unit != null ?
+        providerSchema.unit.toLowerCase() : defaultUnit;
+}
+
+function getMinTemperature(unit) {
+    return unit === 'f' ? 50 : 9;
+}
+
+function getMaxTemperature(unit) {
+    return unit === 'f' ? 90 : 32;
+}
+
+function convertTemperature(temperature, from, to) {
+    if (from === 'c' && to === 'f') {
+        return celsiusToFahrenheit(temperature);
+    } else if (from === 'f' && to === 'c') {
```

Comment:
else not needed

Reply:
else needed - user could call convertTemperatur(72, 'f', 'f')